routes/api/v1/cruises.js

```diff
@@ -573,8 +569,18 @@ exports.plugin = {
             return Boom.notFound('No record found for id: ' + request.params.id);
           }
 
-          if (!request.auth.credentials.scope.includes("admin") && result.cruise_hidden && (useAccessControl && typeof result.cruise_access_list !== 'undefined' && !result.cruise_access_list.includes(request.auth.credentials.id))) {
-            return Boom.unauthorized('User not authorized to retrieve this cruise');
+          // Check if user can access this cruise
+          if (result.cruise_hidden) {
+            // If not authenticated, cannot access hidden cruises
+            if (!request.auth.credentials || !request.auth.credentials.scope) {
+              return Boom.unauthorized('Cannot access hidden cruise without authentication');
+            }
+            // If authenticated but not admin, check access list
+            if (!request.auth.credentials.scope.includes("admin") &&
```

Comment:
Doesn't guard against `credentials` being null as elsewhere -- but I'm again not sure if the guard is necessary.